pkg/deployment/collector_test.go

```diff
@@ -91,6 +92,27 @@ func TestCollectorAnnotations(t *testing.T) {
 	assert.Equal(t, "false", dep.Spec.Template.Annotations["prometheus.io/scrape"])
 }
 
+func TestCollectorSecrets(t *testing.T) {
+	jaeger := v1alpha1.NewJaeger("TestCollectorSecrets")
+	data := map[string]string{
```

Comment:
The data defined here, and referenced in the `Secret` seems redundant - hence my comment above about whether we just need a list of secret names?

Reply:
Right, here it is redundant, because the secret should get converted into the environment variable at container runtime, but in the definition of `JaegerStorageSpec` above, we'll need to SecretList to actually store the actual name and value of the environment variable right? (This block here - https://github.com/jaegertracing/jaeger-operator/pull/114/files#diff-d4532436f335915e1ed831b4bf0c67a9R56 - passes the secret into the environment variable)

Reply:
That code you referenced is linking the whole secret into the container - so all the key value pairs in the secret would be mapped to environment variables (as I understand it).

However using the SecretList type implies you might be expecting users to define the secrets themselves in the CR (and deployed by the Jaeger operator). Do we want this? Or do we want secrets to be managed independent of the operator.

Possibly I am misunderstanding something, as I am not experienced with using secrets, but maybe can be discussed further once @jpkrohling has reviewed.

Reply:
I see what you mean now. I guess I assumed that users would manage secrets through the operator itself. 

Yes, let's wait for the review. :)

Reply:
We might also be able to support both options - so if the data is defined in the secret, then deploy it - but if no data defined, then assume is deployed separately by the user, and therefore just reference it.

Reply:
I see secrets as objects that are used to allow confidential data (passwords, keys, tokens, ...) to be stored in a safe place and managed via tools like Ansible/Chef/Puppet, avoiding the need to store them in plain text in a source code repository.

With that in mind, I don't think the operator should create a secret to store a value that is specified in the CR. I understand *this* issue/PR here as adding support for the Operator to *read* credentials from secrets.

If we ever need to create confidential data in the future (SSL certs, default users, ...), we should consider creating a secret and storing it there, but I don't think it's the case now.

Reply:
I understand. So the ideal way to do this would be what @objectiser suggests - to actually just pass the path_to_the_secret to the pod, possibly through annotations, and mount the _real_ secret through volumes? Though that's not how k8s secrets are designed I guess.

@jpkrohling thoughts on which approach is best suited? I'll change the PR accordingly.

Reply:
The alternative would be to turn `Secrets  v1.SecretList` into `Secrets []string` - so that any secret listed here would be assumed to define a set of key/value pairs that would be the environment variables?

Depends whether we want the CR to explicitly list the parameters (`es.password`) and then do the conversion to equivalent env var name - or instead have the secrets just list the env variables explicitly.

Reply:
> or instead have the secrets just list the env variables explicitly

This would make the content of the secret exposed via `oc get pod ... -o yaml`, which might become a security problem (information leakage) if the user has no access to secrets.

Reply:
The same would be true with the `secretKeyRef` approach - the contents of the secret would still be the same key/value pairs (as far as I can tell - so if wrong can you provide ref to relevant docs).

Reply:
I just talked to @objectiser via IRC and I understand now what was up for discussion here. I agree that the only change required for this one would be to convert the SecretList into a list of strings containing the secret names. Question: should it support secrets in other namespaces?

Reply:
OK. Will fix.